kytos/core/daemon.py

```diff
@@ -1,104 +1,32 @@
-#         print("Terminate requested, without context")
-#         sys.exit(0)
-#
-#
-# class KytosDaemon(object):
-#     """Daemonize and run the kytos daemon."""
-#
-#     def __init__(self, options):
-#         """Constructor of KytosDaemon service the parameters below.
-#
-#         Parameters
-#             options (dict): python dictionary with options to run kytos.
-#         """
-#         self.options = options
-#         context_class = DaemonContext if options.daemon else NoDaemonContext
-#         self.context = self._build_context(options, context_class)
-#
-#     def _build_context(self, options, context_class):
-#         signal_map = {
-#             signal.SIGHUP:  self._handle_reconfigure,
-#             signal.SIGINT:  self._handle_graceful_shutdown,
-#             signal.SIGTERM: self._handle_shutdown,
-#         }
-#         pidfile = PIDLockFile(self.options.pidfile)
-#         return context_class(
-#             working_directory=self.options.workdir,
-#             umask=0x022,
-#             pidfile=pidfile,
-#             signal_map=signal_map,
-#             files_preserve=[pidfile.path])
-#
-#     def run(self):
-#         """Method used to hold kytos processes."""
-#         # setup_logging(self.options)
-#         # We should print nice message if daemon is_open.
-#         with self.context:
-#             while True:
-#                 print(self.context)
-#                 time.sleep(2)
-#
-#     def _handle_shutdown(self, sig_num, stack_frame):
-#         print("Shutdown requested: sig %s" % sig_num)
-#         # self.reactor.stop()
-#         self.context.terminate(sig_num, stack_frame)
-#
-#     def _handle_graceful_shutdown(self, sig_num, stack_frame):
-#         """Gracefully shutdown by waiting for Jobs to finish."""
-#         print("Graceful Shutdown requested: sig %s" % sig_num)
-#         self.context.terminate(sig_num, stack_frame)
-#         self._wait_for_jobs()
-#
-#     def _wait_for_jobs(self):
-#         """Waiting for jobs."""
-#         pass
-#
-#     def _handle_reconfigure(self, _signal_number, _stack_frame):
-#         """Reconfigure requested by SIGHUP."""
-#         # asyncio reconfigure task on call later ?
-#         pass
+from daemons.prefab import run
+
+
+class KytosDaemon(run.RunDaemon):  # pylint: disable=too-many-ancestors
+    """Class to handle kytos controller in daemon mode."""
+
+    def __init__(self, controller=None):
+        """Instance a KytosDaemon.
+
+        Args:
+            controller(KytosController): Kytos controller
+        """
+        self.controller = controller
+        super().__init__(pidfile=controller.options.pidfile)
+
+    def run(self):
+        """Run the controller and keep alive."""
+        try:
+            self.controller.start()
+        except SystemExit as exception:
+            self.controller.log.error(exception)
+            self.controller.log.info("Kytos daemon start aborted.")
+        while True:
+            pass
```

Comment:
Is this `while True` right? When is it going to be 'stopped'?
@cemsbr it looks strange to me, do you agree?

Reply:
This is only to keep the process Alive. When the signal SIGTERM or SIGINT is received the process will call the controller.stop to stop the controller instance and after that, the process will be killed.

Reply:
Ok! Thanks for the explanation!